commit 607d201e22c5f227f4eb26131760d3613e87edf0
Author: Chris Patterson <pattersonc@ainfosec.com>
Date:   Mon Feb 9 16:13:05 2015 -0500

    qemu-device: add disk readonly option and set readonly disks to scsi
    
    qemu's ide does not support readonly=on flag, so set readonly disks
    to scsi.
    
    Signed-off-by: Chris Patterson <pattersonc@ainfosec.com>

diff --git a/src/qemu-device.c b/src/qemu-device.c
index 68f1437..356672f 100644
--- a/src/qemu-device.c
+++ b/src/qemu-device.c
@@ -224,21 +224,32 @@ static bool drive_device_parse_options (struct device_model *devmodel,
     char *media;
     char *format;
     char *index;
+    char *readonly;
     bool res = false;
 
     file = retrieve_option (devmodel, device, "file", drivefile);
     media = retrieve_option (devmodel, device, "media", drivemedia);
     format = retrieve_option (devmodel, device, "format", driveformat);
     index = retrieve_option (devmodel, device, "index", driveindex);
+    readonly = retrieve_option (devmodel, device, "readonly", drivereadonly);
 
     SPAWN_ADD_ARGL (devmodel, end_drive, "-drive");
-    SPAWN_ADD_ARGL (devmodel, end_drive,
+
+    // readonly hard disks are scsi, cdrom and writeable disks are ide
+    if ((strcmp(readonly, "off") == 0) || (strcmp(media, "cdrom") == 0)) {
+        SPAWN_ADD_ARGL (devmodel, end_drive,
                     "file=%s,if=ide,index=%s,media=%s,format=%s",
                     file, index, media, format);
-
+    } else {
+        SPAWN_ADD_ARGL (devmodel, end_drive,
+                    "file=%s,if=scsi,index=%s,media=%s,format=%s,readonly=on",
+                    file, index, media, format);
+    }
     res = true;
 
 end_drive:
+    free (readonly);
+drivereadonly:
     free (index);
 driveindex:
     free (format);
